File: widget/src/main/java/com/yyxnb/widget/action/ClickAction.java
package com.yyxnb.widget.action;

import android.support.annotation.IdRes;
import android.view.View;

/**
 * 点击事件意图
 *
 * @author yyx
 */
public interface ClickAction extends View.OnClickListener {

    <V extends View> V findViewById(@IdRes int id);

    @Override
    default void onClick(View v) {
        // 默认不实现，让子类实现
    }

    default void setOnClickListener(@IdRes int... ids) {
        for (int id : ids) {
            findViewById(id).setOnClickListener(this);
        }
    }

    default void setOnClickListener(View... views) {
        for (View view : views) {
            view.setOnClickListener(this);
        }
    }
}

File: widget/src/main/java/com/yyxnb/widget/action/ClickAction.kt
package com.yyxnb.widget.action

import android.view.View
import androidx.annotation.IdRes

/**
 * 点击事件意图
 *
 * @author yyx
 */
interface ClickAction : View.OnClickListener {

    fun <V : View> findViewById(@IdRes id: Int): V

    override fun onClick(v: View) {
        // 默认不实现，让子类实现
    }

    fun setOnClickListener(@IdRes vararg ids: Int) {
        for (id in ids) {
            findViewById<View>(id).setOnClickListener(this)
        }
    }

    fun setOnClickListener(vararg views: View) {
        for (view in views) {
            view.setOnClickListener(this)
        }
    }
}